nixos/tests/endlessh-go.nix

```
@@ -0,0 +1,58 @@
+import ./make-test-python.nix ({ lib, pkgs, ... }:
+{
+  name = "endlessh-go";
+  meta.maintainers = with lib.maintainers; [ azahi ];
+
+  nodes = {
+    server = { ... }: {
+      services.endlessh-go = {
+        enable = true;
+        prometheus.enable = true;
+        openFirewall = true;
+      };
+
+      specialisation = {
+        unprivileged.configuration = {
+          services.endlessh-go = {
+            port = 2222;
+            prometheus.port = 9229;
+          };
+        };
+
+        privileged.configuration = {
+          services.endlessh-go = {
+            port = 22;
+            prometheus.port = 922;
+          };
+        };
+      };
+    };
+
+    client = { pkgs, ... }: {
+      environment.systemPackages = with pkgs; [ curl netcat ];
+    };
+  };
+
+  testScript = { nodes, ... }: ''
+    def activate_specialisation(name: str):
+        server.succeed(f"${nodes.server.config.system.build.toplevel}/specialisation/{name}/bin/switch-to-configuration test >&2")
+
+    start_all()
+
+    with subtest("Unprivileged"):
+        activate_specialisation("unprivileged")
+        server.wait_for_unit("endlessh-go.service")
+        server.wait_for_open_port(2222)
+        server.wait_for_open_port(9229)
+        client.succeed("nc -z -v server 2222")
```

I wasn't able to use `dd` here because it hanged infinitely for some reason. Any suggestions on how to actually tests that garbage bytes are received would be very much appreciated.

---

```suggestion
        client.succeed("nc -dv server 2222 | dd bs=1 count=2")
```
seems to work, but apparently endlessh doesn't send only one byte per internval, so
```suggestion
        client.succeed("nc -dvW5 server 2222")
```
is probably better